runtime/common/src/weight_params.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+/// Generates Weight params (Get impl) for Runtimes with the max proof size pulled from the relay if in an externalities provided
+/// environment, and the max proof size for relay chains as defined by polkadot used if not.
+/// Provides:
+/// - MaximumBlockWeight: MAXIMIM_BLOCK_WEIGHT with proof size adjusted for relay chain val.
+/// - BlockWeightsWithRelayProof: BlockWeights generated with using MaximumBlockWeight with relay proof size set.
+/// - MessagingReservedWeight: chain messaging reserved weight using MaximumBlockWeight with relay proof size set.
+/// - MaximumSchedulerWeight: max scheduler weight using MaximumBlockWeight with relay proof size set.
+
+#[macro_export]
+macro_rules! gen_weight_parameters {
+	() => {
+		//  Currently using a Macro as seperating these out into a seperate module usable by the different runtimes
+		// causes issues with the `__construct_runtime_integrity_test::runtime_integrity_tests` test for runtime/development
+		// with the cfg!(test) not returning correctly for seperate module, and then attempted data fetch relay chain with no
+		// externalities provided... which panics.  Works for tests with provided externalities env & execute_with
+
+		// Curently have Runtime generic as we would need this to switch back from macro, though we could technically
+		// remove now, though the impl where does provide context for which config would need to be impl for runtime
+		// to fetch the validation data for the relay max proof size.
+
+		/// Struct for Get impl of MaximumBlockWeight with Weight using relay max_pov_size as proof size.
+		pub struct MaximumBlockWeight<Runtime>(sp_std::marker::PhantomData<Runtime>);
+
+		impl<Runtime> sp_core::Get<frame_support::weights::Weight> for MaximumBlockWeight<Runtime>
+		where
+			Runtime: cumulus_pallet_parachain_system::Config,
+		{
+			fn get() -> frame_support::weights::Weight {
+				if cfg!(test) {
```

Comment:
I guess the need for the macro usage was caused by the fact that only the compiled crate is compiling with `test`. So if you compile i.e. `development-runtime` with `test` then `runtime-common` (where this lives) will not compile with `test`. 

I think we can avoid to use the macro if we check against `std` instead.

Another option is to get this value through a function where the `cfg(test)` value is passed by param.